python/pyiceberg/utils/file_stats.py

```diff
@@ -0,0 +1,333 @@
+        col_path_2_iceberg_id: A mapping of column paths as in the `path_in_schema` attribute of the colum
+            metadata to iceberg schema IDs. For scalar columns this will be the column name. For complex types
+            it could be something like `my_map.key_value.value`
+        file_size (int): The total compressed file size cannot be retrieved from the metadata and hence has to
+            be passed here. Depending on the kind of file system and pyarrow library call used, different
+            ways to obtain this value might be appropriate.
+    """
+    col_index_2_id = {}
+
+    col_names = set(metadata.schema.names)
+
+    first_group = metadata.row_group(0)
+
+    for c in range(metadata.num_columns):
+        column = first_group.column(c)
+        col_path = column.path_in_schema
+
+        if col_path in col_path_2_iceberg_id:
+            col_index_2_id[c] = col_path_2_iceberg_id[col_path]
+        else:
+            raise AssertionError(f"Column path {col_path} couldn't be mapped to an iceberg ID")
+
+    column_sizes: Dict[int, int] = {}
+    value_counts: Dict[int, int] = {}
+    split_offsets: List[int] = []
+
+    null_value_counts: Dict[int, int] = {}
+    nan_value_counts: Dict[int, int] = {}
+
+    col_aggs = {}
+
+    for r in range(metadata.num_row_groups):
+        # References:
+        # https://github.com/apache/iceberg/blob/fc381a81a1fdb8f51a0637ca27cd30673bd7aad3/parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java#L232
+        # https://github.com/apache/parquet-mr/blob/ac29db4611f86a07cc6877b416aa4b183e09b353/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/metadata/ColumnChunkMetaData.java#L184
+
+        row_group = metadata.row_group(r)
+
+        data_offset = row_group.column(0).data_page_offset
+        dictionary_offset = row_group.column(0).dictionary_page_offset
+
+        if row_group.column(0).has_dictionary_page and dictionary_offset < data_offset:
+            split_offsets.append(dictionary_offset)
+        else:
+            split_offsets.append(data_offset)
+
+        for c in range(metadata.num_columns):
+            col_id = col_index_2_id[c]
+
+            column = row_group.column(c)
+
+            column_sizes[col_id] = column_sizes.get(col_id, 0) + column.total_compressed_size
+            value_counts[col_id] = value_counts.get(col_id, 0) + column.num_values
+
+            if column.is_stats_set:
+                try:
+                    statistics = column.statistics
+
+                    null_value_counts[col_id] = null_value_counts.get(col_id, 0) + statistics.null_count
+
+                    if column.path_in_schema in col_names:
+                        # Iceberg seems to only have statistics for scalar columns
+
+                        if col_id not in col_aggs:
+                            col_aggs[col_id] = StatsAggregator(statistics.physical_type)
+
+                        col_aggs[col_id].add_min(statistics.min)
```

Comment:
Do we need the intermediate `col_aggs`? I would prefer to directly write it into `{lower,upper}_bounds`.

Reply:
Fair point, thanks for the context